src/firefly/js/ui/tap/TapSearchRootPanel.jsx

```diff
@@ -424,6 +424,17 @@ function getAdqlQuery(tapBrowserState, showErrors= true) {
     const tableName = maybeQuote(tapBrowserState?.tableName, true);
     if (!tableName) return;
     const isUpload= isTapUpload(tapBrowserState);
+
+    if (isUpload) { //check for more than one upload file (in Spatial and in ObjectID col) - should this be a utility function in constraints.js?
+        const { constraintFragments } = tapBrowserState;
+        const entries = [...constraintFragments.values()];
+        const matchingEntries = entries.filter((c) => Boolean(c.uploadFile && c.TAP_UPLOAD && c.adqlConstraint));
+        if (matchingEntries.length > 1) {
+            if (showErrors) showInfoPopup('More than one upload is not supported at this moment.', 'Error');
+            return;
+        }
+    }
+
```

Comment:
Does it make sense to have this here? or should it be a utility function in constraints.js? (similar to isTapUpload on line 426 - or I could even combine the 2 and return an object with 2 entries: one would indicate true/false for isTapUpload, and the other would include the number of matching entries (so more than 1 upload would be flagged as false)